src/ray/core_worker/task_manager.cc

```diff
@@ -222,18 +228,18 @@ void TaskManager::CompletePendingTask(const TaskID &task_id,
             return_object.metadata().size());
       }
 
-      std::vector<ObjectID> nested_ids;
-      nested_ids.reserve(return_object.nested_refs().size());
-      for (const auto &nested_ref : return_object.nested_refs()) {
-        nested_ids.push_back(ObjectID::FromBinary(nested_ref.object_id()));
-      }
-
       bool stored_in_direct_memory = in_memory_store_->Put(
           RayObject(data_buffer, metadata_buffer, nested_ids), object_id);
       if (stored_in_direct_memory) {
         direct_return_ids.push_back(object_id);
       }
     }
+
+    rpc::Address owner_address;
+    if (reference_counter_->GetOwner(object_id, &owner_address) &&
+        reference_counter_->OwnedByUs(object_id)) {
```

Comment:
I don't think we need these if checks since we will always own an object returned by a task we called.

Maybe just check if `!nested_ids.empty()`?